smalltalksrc/Slang-Tests/CSlangPrettyPrinterTest.class.st

```diff
@@ -155,10 +153,10 @@ CSlangPrettyPrinterTest >> testVisitCompoundStatement [
 	print := self prettyPrint: (CCompoundStatementNode
 			  declarations: { (CAssignmentNode
 						lvalue: (CIdentifierNode name: 'toto')
-						rvalue: (CConstantNode value: '99')) }
+						rvalue: (CConstantNode value: '99')) withoutParentheses }
```

Comment:
These `withoutParentheses` were added because the expressions have parentheses by default? Was not better to change the default? 🤔

Reply:
Actually, expressions have no parentheses by default, but assignments do.
What I did here is to decide if a child needs parentheses in the parent node, which is better than a global strategy.
However, the precedence in C is complex (https://en.cppreference.com/w/c/language/operator_precedence) so:
 - the simplest is to parenthesize everything. Uglier but correct code
 - the hardest is to embed C precedence into the pretty printer and put parentheses only if necessary (i.e., if the parent has less precedence than the child in the tree)